tests/unit/ImplicitAnimationTests.swift

```diff
@@ -0,0 +1,186 @@
+    }
+    do {
+      let animation = addedAnimations[1] as! CABasicAnimation
+      XCTAssertFalse(animation.isAdditive)
+      XCTAssertEqual(animation.keyPath, AnimatableKeyPath.position.rawValue)
+      XCTAssertEqual(animation.fromValue as! CGPoint, .init(x: 0, y: 0))
+      XCTAssertEqual(animation.toValue as! CGPoint, .init(x: 50, y: 50))
+      XCTAssertEqual(animation.duration, timing.duration)
+
+      let addedCurve = MotionCurve(fromTimingFunction: animation.timingFunction!)
+      XCTAssertEqual(addedCurve.type, timing.curve.type)
+      XCTAssertEqual(addedCurve.data.0, timing.curve.data.0)
+      XCTAssertEqual(addedCurve.data.1, timing.curve.data.1)
+      XCTAssertEqual(addedCurve.data.2, timing.curve.data.2)
+      XCTAssertEqual(addedCurve.data.3, timing.curve.data.3)
+    }
+  }
+
+  func testFrameActionAddsTwoAnimations() {
+    animator.animate(with: timing) {
+      self.view.frame = .init(x: 0, y: 0, width: 100, height: 100)
+    }
+
+    XCTAssertEqual(addedAnimations.count, 2)
+
+    do {
+      let animation = addedAnimations
+          .flatMap { $0 as? CABasicAnimation }
+          .first(where: { $0.keyPath == AnimatableKeyPath.position.rawValue})!
+      XCTAssertFalse(animation.isAdditive)
+      XCTAssertEqual(animation.fromValue as! CGPoint, .init(x: 0, y: 0))
+      XCTAssertEqual(animation.toValue as! CGPoint, .init(x: 50, y: 50))
+      XCTAssertEqual(animation.duration, timing.duration)
+
+      let addedCurve = MotionCurve(fromTimingFunction: animation.timingFunction!)
+      XCTAssertEqual(addedCurve.type, timing.curve.type)
+      XCTAssertEqual(addedCurve.data.0, timing.curve.data.0)
+      XCTAssertEqual(addedCurve.data.1, timing.curve.data.1)
+      XCTAssertEqual(addedCurve.data.2, timing.curve.data.2)
+      XCTAssertEqual(addedCurve.data.3, timing.curve.data.3)
+    }
+    do {
+      let animation = addedAnimations
+        .flatMap { $0 as? CABasicAnimation }
+        .first(where: { $0.keyPath == "bounds"})!
+      XCTAssertFalse(animation.isAdditive)
+      XCTAssertEqual(animation.fromValue as! CGRect, .init(x: 0, y: 0, width: 0, height: 0))
+      XCTAssertEqual(animation.toValue as! CGRect, .init(x: 0, y: 0, width: 100, height: 100))
+      XCTAssertEqual(animation.duration, timing.duration)
+
+      let addedCurve = MotionCurve(fromTimingFunction: animation.timingFunction!)
+      XCTAssertEqual(addedCurve.type, timing.curve.type)
+      XCTAssertEqual(addedCurve.data.0, timing.curve.data.0)
+      XCTAssertEqual(addedCurve.data.1, timing.curve.data.1)
+      XCTAssertEqual(addedCurve.data.2, timing.curve.data.2)
+      XCTAssertEqual(addedCurve.data.3, timing.curve.data.3)
+    }
+  }
+
+  func testOneActionAddsNoAnimationWhenActionsDisable() {
+    CATransaction.begin()
+    CATransaction.setDisableActions(true)
+
+    animator.animate(with: timing) {
+      self.view.alpha = 0
+    }
+
+    CATransaction.commit()
+
+    XCTAssertEqual(addedAnimations.count, 0)
```

Comment:
Also XCTAssert the view's alpha is immediately set?

Reply:
Done.